### PR TITLE
add new class "att.predicate"

### LINKED ISSUE #1925: <equiv> should get the @predicate attribute
The `<equiv>` element should get the same or similarly defined `@predicate` attribute that is available on `<model>`.  This would enable filtered equivalences for all sorts of uses, but for example being able to map uses of a single element to different conceptual models. i.e. `<name type="person">` could map to one concept in (say) CIDOC CRM or other ontologies but `<name type="place">` could map to another. These would be documented in the ODD not in the document instance.  This strikes me as a very flexible addition to equiv that is easy to implement.

### PR DESCRIPTION
**Pull request by peterstadler**
* move `@predicate` from `<model>` to the new class
* make `<equiv>` a member of this new class as requested in #1925

### CODE REVIEW

File: P5/Source/Specs/att.predicate.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            concepts of "PERSON" and "PLACE" depending on the XPATH expression given in <att>predicate</att></p>
+        <egXML xmlns="http://www.tei-c.org/ns/Examples" source="#NONE">
+            <elementSpec ident="name" mode="change">
+                <equiv name="PERSON" predicate="@type='person'"
+                    uri="http://www.example.com/entities/person"/>
+                <equiv name="PLACE" predicate="@type='place'"
+                    uri="http://www.example.com/entities/place"/>
+            </elementSpec>
+        </egXML>
+    </exemplum>
```
Comment (sydb) [inline P5/Source/Specs/att.predicate.xml:45]: Probably a good idea to provide a `<listRef>` here. Off the cuff, I think it should be
~~~xml
     <listRef>
       <ptr target="#TDcrystalsCE"/>
       <ptr target="#TDPMPM"/>
     </listRef>
~~~

File: P5/Source/Specs/att.predicate.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            <model predicate="parent::person"
+                behaviour="inline">
+                <desc versionDate="2015-08-21"
+                    xml:lang="en">If it is a child of a person element, treat as
+                    inline</desc>
+            </model>
+        </egXML>
+    </exemplum>
+    <exemplum xml:lang="en">
+        <p>The following example declares that the <gi>name</gi> element can be mapped to (is equivalent to) the external
```
Comment (sydb) [inline P5/Source/Specs/att.predicate.xml]: I think I would mildly prefer 
~~~xml
<p>The following example declares that the <gi>name</gi> element can be mapped to, or is equivalent to, the external
~~~

File: P5/Source/Specs/att.predicate.xml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                behaviour="inline">
+                <desc versionDate="2015-08-21"
+                    xml:lang="en">If it is a child of a person element, treat as
+                    inline</desc>
+            </model>
+        </egXML>
+    </exemplum>
+    <exemplum xml:lang="en">
+        <p>The following example declares that the <gi>name</gi> element can be mapped to (is equivalent to) the external 
+            concepts of "PERSON" and "PLACE" depending on the XPATH expression given in <att>predicate</att></p>
```
Comment (sydb) [inline P5/Source/Specs/att.predicate.xml]: I’m not sure, but I think "PERSON" and "PLACE" should perhaps be `<mentioned>PERSON</mentioned>` and `<mentioned>PLACE</mentioned>`. (Or maybe `<val>` or `<soCalled>` or `<q>`.)
But I am sure that “XPATH” should be “XPath”.

File: P5/Source/Specs/att.predicate.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        <egXML xmlns="http://www.tei-c.org/ns/Examples" source="#NONE">
+            <elementSpec ident="name" mode="change">
+                <equiv name="PERSON" predicate="@type='person'"
```
Comment (sydb) [inline P5/Source/Specs/att.predicate.xml]: `@type eq 'person'` ?

File: P5/Source/Specs/att.predicate.xml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                    uri="http://www.example.com/entities/person"/>
+                <equiv name="PLACE" predicate="@type='place'"
```
Comment (sydb) [inline P5/Source/Specs/att.predicate.xml]: `@type eq 'place'` ?

**Review by sydb — CHANGES_REQUESTED** [review]
I think somewhere it should be explicitly mentioned that the value of `@predicate` is an XPath expression _without the square brackets_.

File: P5/Source/Specs/att.predicate.xml (hunk 4 above)
Comment (tuurma) [reply P5/Source/Specs/att.predicate.xml]: doesn't really matter here, no?

Comment (sydb) [reply P5/Source/Specs/att.predicate.xml]: In the sense that both `=` and `eq` will always yield the same results, no, it does not matter. In the sense that `@type` of `<name>` and the literal `'person`' are both, by definition, singular, it is suboptimal to be using a sequence operator.

File: P5/Source/Specs/att.predicate.xml (hunk 1 above)
Comment (peterstadler) [reply P5/Source/Specs/att.predicate.xml:45]: addressed in 116f2b4c6

File: P5/Source/Specs/att.predicate.xml (hunk 2 above)
Comment (peterstadler) [reply P5/Source/Specs/att.predicate.xml]: addressed in 0a5a314ed

File: P5/Source/Specs/att.predicate.xml (hunk 3 above)
Comment (peterstadler) [reply P5/Source/Specs/att.predicate.xml]: addressed in 0a5a314

File: P5/Source/Specs/att.predicate.xml (hunk 4 above)
Comment (peterstadler) [reply P5/Source/Specs/att.predicate.xml]: addressed in 4c0fca7b7

File: P5/Source/Specs/att.predicate.xml (hunk 5 above)
Comment (peterstadler) [reply P5/Source/Specs/att.predicate.xml]: addressed in 4c0fca7b7

**Comment by peterstadler** [comment]
addressed @sydb 's concerns, please have a look (again)

cc'ing @jamescummings

**Review by tuurma — APPROVED** [review]

**Review by sydb — APPROVED** [review]
Do it, Chewie!

**Pull request merged**